otx/algorithms/detection/adapters/mmdet/datasets/dataset.py

```diff
@@ -362,14 +342,17 @@ def __init__(
             iou_threshold=iou_threshold,
             max_per_img=max_per_img,
             max_annotation=max_annotation,
-            filter_empty_gt=filter_empty_gt if self.dataset.otx_dataset[0].subset != Subset.TESTING else False,
-            sampling_ratio=sampling_ratio if self.dataset.otx_dataset[0].subset != Subset.TESTING else 1.0,
-            include_full_img=include_full_img if self.dataset.otx_dataset[0].subset != Subset.TESTING else True,
+            filter_empty_gt=filter_empty_gt if data_subset != Subset.TESTING else False,
+            sampling_ratio=sampling_ratio if data_subset != Subset.TESTING else 1.0,
+            include_full_img=include_full_img if data_subset != Subset.TESTING else True,
         )
         self.flag = np.zeros(len(self), dtype=np.uint8)
         self.pipeline = Compose(pipeline)
         self.test_mode = test_mode
         self.num_samples = len(self.dataset)  # number of original samples
+        annotation = [self.get_ann_info(i) for i in range(len(self))]
+        if data_subset == Subset.VALIDATION:
```

Comment:
It's just a queestion. I think `self.evaluator` is initialized only if `data_subset == Subset.VALIDATION` is `True` but don't other cases  something like `Subset.TESTING` need `self.evaluator`?

Reply:
The evaluation function had some issues when given empty ground truth annotation. I removed this condition and changed a bit in the evaluator so now it takes empty gt without outputting errors